mesonbuild/modules/python.py

```diff
@@ -232,6 +232,11 @@ def extension_module_method(self, args: T.Tuple[str, T.List[BuildTargetSource]],
         kwargs['name_prefix'] = ''
         kwargs['name_suffix'] = target_suffix
 
+        # In AIX if we are using a extension module, this means a python module does not
+        # require a archived shared library due to dlopen mechanism of python. Hence do not archive.
+        if mesonlib.is_aix():
+            SharedLibrary.aix_so_archive = False
```

Comment:
We should avoid mutating global state like this.

Reply:
I think we can just get rid of it altogether because we:
```python
        return self.interpreter.build_target(self.current_node, args, kwargs, SharedModule)
```

So we are already handling a SharedModule, and we *should* already be getting skipped due to the condition:
```python
if isinstance(target, build.SharedLibrary) and not isinstance(target, build.SharedModule)
```

Reply:
You are right here. Yes, we should remove this.